packages/block-editor/src/components/editor-styles/index.js

```diff
@@ -0,0 +1,38 @@
+/**
+ * External dependencies
+ */
+import { compact, map } from 'lodash';
+
+/**
+ * WordPress dependencies
+ */
+import { useEffect } from '@wordpress/element';
+
+/**
+ * Internal dependencies
+ */
+import transformStyles from '../../utils/transform-styles';
+
+function EditorStyles( { styles } ) {
+	useEffect( () => {
+		const updatedStyles = transformStyles(
+			styles,
+			'.editor-styles-wrapper'
+		);
+
+		const nodes = map( compact( updatedStyles ), ( updatedCSS ) => {
+			const node = document.createElement( 'style' );
+			node.innerHTML = updatedCSS;
+			document.body.appendChild( node );
+
+			return node;
+		} );
+
+		return () =>
+			nodes.forEach( ( node ) => document.body.removeChild( node ) );
```

Comment:
Related to https://github.com/WordPress/gutenberg/pull/20982#issuecomment-604582993, I observe this unmounting behavior is "new", though again not entirely clear what impact (if any) it would have.

One worry I might have is that since this unsubscribe behavior is called if any dependencies change, are we sure that `styles` will remain stable? Aside from my current debugging of test failures, seems like it might be a performance concern if the references change with any frequency.

Reply:
> are we sure that styles will remain stable

I think so since these come from the server. It would surprise if this is the reason for the failure.

The reason i added the unmount behavior is because it's the right thing to do if the styles changes, we need to recompute and reload them.